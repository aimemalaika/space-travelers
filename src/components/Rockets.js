import { useDispatch, useSelector } from 'react-redux';
import { reserveRocket, cancelReservation } from '../redux/rockets/rockets';

const Rockets = () => {
  const rockets = useSelector((state) => state.reducerRockets);

  const dispatch = useDispatch();

  const reserveCancelRocket = (id, status) => {
    if (status === true) {
      dispatch(cancelReservation(id));
    } else {
      dispatch(reserveRocket(id));
    }
  };

  const btnReserve = {
    color: 'white',
    backgroundColor: '#007bff',
    border: 'none',
  };

  const btnCancelation = {
    color: '#6d757d',
    backgroundColor: 'white',
    border: '1px solid #6d757d',
  };

  const badge = {
    backgroundColor: '#18a2b8',
    color: '#fff',
    padding: '2px 5px',
    borderRadius: '5px',
    marginRight: '10px',
  };

  return (
    <div className="rockets-list">
      {rockets.map((rock) => (
        <div className="rocket-element" key={rock.id}>
          <div className="image-content">
            <img src={rock.flickr_images[0]} alt={rock.rocket_name} />
          </div>

          <div className="rocket-info">
            <h3>{rock.rocket_name}</h3>
            <p>
              {rock.reserved && (
                <span style={rock.reserved && badge}>Reserved</span>
              )}
              {rock.description}
            </p>
            <button
              style={(!rock.reserved && btnReserve) || (rock.reserved && btnCancelation)}
              type="submit"
              onClick={() => reserveCancelRocket(rock.id, rock.reserved)}
            >
              {(!rock.reserved && 'Reserve Rockets') || (rock.reserved && 'Cancel Reservation')}
            </button>
          </div>
        </div>
      ))}
    </div>
  );
};

export default Rockets;
