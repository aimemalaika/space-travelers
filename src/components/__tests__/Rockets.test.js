import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Rockets from '../Rockets';
import store from '../../redux/configureStore';

describe('Test the Rockets list', () => {
  it('Rockets section renders correctly', () => {
    const rockets = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    ).toJSON();

    expect(rockets).toMatchSnapshot();
  });
});
