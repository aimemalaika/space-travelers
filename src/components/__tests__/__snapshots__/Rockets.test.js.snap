// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Rockets list Rockets section renders correctly 1`] = `
<div
  className="rockets-list"
/>
`;
